tests/gis_tests/layermap/tests.py

```diff
@@ -319,7 +319,10 @@ def db_for_write(self, model, **hints):
         return self.db_for_read(model, **hints)
 
     def allow_relation(self, obj1, obj2, **hints):
-        return None
+        # Required after fixing issue #27629
+        # ContentType objects are created during a post-migrate signal while performing fixture teardown
+        # using the default database alias and do not abide by the database specified by this router.
+        return True
```

Comment:
After some digging, it appears that `django/contrib/auth/management/__init__.py` `create_permissions()` is creating some `ContentType` objects using the default database alias (by specifying it to the db_manager) and not 'other' specified by the router. Some related objects are being created in the 'other' database, and the relation check is then failing since they exist in different databases.

I would argue the proper fix is to make the `create_permissions()` function use the routers properly. Short of that, this seems like the least invasive solution.

But, this still makes me worry that other systems in the wild may run into this hidden bug once this fix is applied.

Reply:
I'll investigate this later.